src/App.tsx

```diff
@@ -1,14 +1,82 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [search, setSearch] = useState<string>('');
+  const [status, setStatus] = useState<string>('all');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [updatedAt, setUpdatedAt] = useState(new Date());
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Failed to load todos');
+      });
+  }, [updatedAt]);
+
+  const filteredTodos = useMemo(() => {
+    if (!todos) {
+      return [];
+    }
+
+    return todos.filter(todo => {
+      if (status === 'all') {
+        return todo.title.toLowerCase().includes(search.toLowerCase());
+      }
+
+      return (
+        todo.title.toLowerCase().includes(search.toLowerCase()) &&
+        todo.completed === (status === 'completed')
+      );
+    });
+  }, [todos, status, search]);
+
+  const reload = () => {
+    setUpdatedAt(new Date());
+    setErrorMessage('');
+  };
+
+  const getTodosListView = () => {
+    if (!todos && !errorMessage) {
+      return <Loader />;
+    }
+
+    if (todos && todos?.length > 0) {
+      return (
+        <TodoList
+          todos={filteredTodos}
+          onSelect={setSelectedTodo}
+          selectedTodo={selectedTodo}
+        />
+      );
+    }
+
+    if (errorMessage && todos?.length === 0) {
```

Comment:
The condition `if (errorMessage && todos?.length === 0)` might not correctly handle the case when there is an error message but no todos. Consider changing it to `if (errorMessage)` to ensure the error message is displayed whenever there is an error, regardless of the todos' state.